ml-algorithms/src/test/java/org/opensearch/ml/engine/utils/FileUtilsTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.utils;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class FileUtilsTest {
+    private final TemporaryFolder tempDir = new TemporaryFolder();
+
+    @Before
```

Comment:
This @Before, @After could be replaced with 

```
@Rule
    public TemporaryFolder tempDir = new TemporaryFolder();
```

But if you want to stick with manual resource creation and clean up, may be we can do:

```
private TemporaryFolder tempDir;

@Before
public void setUp() throws Exception {
    tempDir = new TemporaryFolder();
    tempDir.create();
}

@After
public void tearDown() {
    if (tempDir != null) {
        tempDir.delete();
    }
}
```

I prefer `tearDown` over `tearUp` which is more conventional.

Reply:
SG, setUp/tearDown is my preference as well - tearUp was copy/paste error. Updated and pushed.